diff -urN nagios-notify-0.14.org/templates/notify-host-by-email.tmpl nagios-notify-0.14/templates/notify-host-by-email.tmpl
--- nagios-notify-0.14.org/templates/notify-host-by-email.tmpl	2012-12-14 11:48:27.000000000 +0100
+++ nagios-notify-0.14/templates/notify-host-by-email.tmpl	2018-02-21 20:49:01.969994349 +0100
@@ -4,6 +4,7 @@
 In-Reply-To: $DATE$@$HOSTNAME$.nagios
 To: "$(encode_mime_header:iso8859-1,$CONTACTALIAS$)" <$CONTACTEMAIL$>
 Subject: Host $HOSTSTATE$ alert for $HOSTNAME$!
+Auto-Submitted: auto-generated
 
 ***** Nagios/$VERSION$ *****
 
diff -urN nagios-notify-0.14.org/templates/notify-service-by-email-conditions.tmpl nagios-notify-0.14/templates/notify-service-by-email-conditions.tmpl
--- nagios-notify-0.14.org/templates/notify-service-by-email-conditions.tmpl	2012-12-14 11:48:27.000000000 +0100
+++ nagios-notify-0.14/templates/notify-service-by-email-conditions.tmpl	2018-02-21 20:49:12.600313242 +0100
@@ -4,6 +4,7 @@
 In-Reply-To: $SERVICEDESC$.$DATE$@$HOSTALIAS$.nagios
 To: "$(encode_mime_header:utf-8,$CONTACTALIAS$)" <$CONTACTEMAIL$>
 Subject: ** $NOTIFICATIONTYPE$ alert - $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **
+Auto-Submitted: auto-generated
 
 ***** Nagios/$VERSION$ *****
 
diff -urN nagios-notify-0.14.org/templates/notify-service-by-email.tmpl nagios-notify-0.14/templates/notify-service-by-email.tmpl
--- nagios-notify-0.14.org/templates/notify-service-by-email.tmpl	2012-12-14 11:48:27.000000000 +0100
+++ nagios-notify-0.14/templates/notify-service-by-email.tmpl	2018-02-21 20:49:19.157176605 +0100
@@ -4,6 +4,7 @@
 In-Reply-To: $SERVICEDESC$.$DATE$@$HOSTALIAS$.nagios
 To: "$(encode_mime_header:iso8859-1,$CONTACTALIAS$)" <$CONTACTEMAIL$>
 Subject: ** $NOTIFICATIONTYPE$ alert - $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **
+Auto-Submitted: auto-generated
 
 ***** Nagios/$VERSION$ *****
 
