diff -urN nagios-notify-0.14.org/nagios-notify nagios-notify-0.14/nagios-notify
--- nagios-notify-0.14.org/nagios-notify	2021-05-04 09:12:22.586849116 +0200
+++ nagios-notify-0.14/nagios-notify	2021-05-04 09:29:15.795337592 +0200
@@ -265,7 +265,7 @@
 		while (match($0, /\$\(base64:([^)]+)\)/)) {
 			pos = length("$(base64:")
 			file = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			$0 = left base64(file) right;
 		}
@@ -274,7 +274,7 @@
 		while (match($0, /\$\(encode_mime_header:([^,]*,[^)]+)\)/)) {
 			pos = length("$(encode_mime_header:")
 			params = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			charset = substr(params, 1, index(params, ",") - 1)
 			data = substr(params, index(params, ",") + 1)
@@ -285,7 +285,7 @@
 		while (match($0, /\$\(timestamp_date:([^)]+)\)/)) {
 			pos = length("$(timestamp_date:")
 			data = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			$0 = left timestamp_date(data) right;
 		}
@@ -294,7 +294,7 @@
 		while (match($0, /\$\(urlencode:([^)]+)\)/)) {
 			pos = length("$(urlencode:")
 			data = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			$0 = left urlencode(data) right;
 		}
@@ -302,7 +302,7 @@
 		while (match($0, /\$\(unescape:([^)]*)\)/)) {
 			pos = length("$(unescape:")
 			data = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			$0 = left unescape(data) right;
 		}
@@ -311,7 +311,7 @@
 		while (match($0, "\$\(math:("MATH_EXPR") *("MATH_OPS") *("MATH_EXPR")\)")) {
 			pos = length("$(math:")
 			expr = substr($0, RSTART + pos, RLENGTH - pos - 1);
-			left = substr($0, 0, RSTART);
+			left = substr($0, 0, RSTART - 1);
 			right = substr($0, RSTART + RLENGTH);
 			$0 = left math(expr) right;
 		}
